index.rst

```diff
@@ -556,6 +556,67 @@ may not serve in place of monkey patching the file: this caching feature may
 need to be optional (if the temporary file storage is getting too big), and it
 is possible that objects could still exist after the SwiftStorage is destroyed.
 
+Formatter
+---------
+
+A Formatter is an object that has the ability to serialize and deserialize
+objects to and from a type of persistent storage. ``StorageInterface``
+subclasses should use formatters to serialize & deserialize objects.
+
+A Formatter is registered for a class type with the ``StorageInterface``
+subclass that should use the formatter to serialize and deserialize instances of
+that class.
+
+The API of the Formatter is only dependent on the ``StorageInterface`` subclass
+that will be calling it. The signature of functions called by the
+``StorageInterface`` subclass should depend on what parameters are required to
+read from and write to the persistent storage type.
```

Comment:
Do you have a Formatter abstract base class for each StorageInterface implementation (I know it's Python and you don't need ABC but it may still be useful)?

Reply:
I don't but that may be something worth adding.